modules/actor/actor-wfrp4e.js

```diff
@@ -1283,125 +1283,155 @@ export default class ActorWfrp4e extends WFRP4eDocumentMixin(Actor)
-      }
-    }
-    else {
-      if (Number.isNumeric(d.system.duration.value)) {
+    let d = foundry.utils.duplicate(disease);
+    let type = d.system.duration.active ? 'duration' : 'incubation';
 
-        d.system.duration.value--
-        if (d.system.duration.value == 0)
-          this.finishDisease(d)
-      }
-      else {
-        let chatData = game.wfrp4e.utility.chatDataSetup(`Attempted to decrement ${d.name} duration but value is non-numeric`, "gmroll", false)
-        chatData.speaker = { alias: this.name }
-        ChatMessage.create(chatData)
+    if (Number.isNumeric(d.system[type].value)) {
+      d.system[type].value--;
+
+      if (d.system[type].value <= 0) {
+        d.system[type].value = 0;
+
+        if (type === 'incubation')
+          await actor.activateDisease(d);
+
+        if (type === 'duration')
+          await actor.finishDisease(d);
       }
+    } else {
+      let chatData = game.wfrp4e.utility.chatDataSetup(`Attempted to decrement ${d.name} ${type} but value is non-numeric`, "gmroll", false);
+      chatData.speaker = {alias: this.name};
+      ChatMessage.create(chatData);
     }
-    this.updateEmbeddedDocuments("Item", [d])
+
+    await this.updateEmbeddedDocuments("Item", [d])
   }
 
+  /**
+   *
+   * @param {ItemWfrp4e} disease
+   *
+   * @return {Promise<void>}
+   */
   async activateDisease(disease) {
     disease.system.duration.active = true;
     disease.system.incubation.value = 0;
-    let msg = game.i18n.format("CHAT.DiseaseIncubation", { disease: disease.name })
+    let msg = game.i18n.format("CHAT.DiseaseIncubation", { disease: disease.name });
+
     try {
-      let durationRoll = (await new Roll(disease.system.duration.value).roll()).total
-      msg += game.i18n.format("CHAT.DiseaseDuration", { duration: durationRoll, unit: disease.system.duration.unit })
+      let durationRoll = (await new Roll(disease.system.duration.value).roll()).total;
+      msg += game.i18n.format("CHAT.DiseaseDuration", { duration: durationRoll, unit: disease.system.duration.unit });
       disease.system.duration.value = durationRoll;
-    }
-    catch (e) {
-      msg += game.i18n.localize("CHAT.DiseaseDurationError")
+    } catch (e) {
+      msg += game.i18n.localize("CHAT.DiseaseDurationError");
     }
 
-    let chatData = game.wfrp4e.utility.chatDataSetup(msg, "gmroll", false)
-    chatData.speaker = { alias: this.name }
-    ChatMessage.create(chatData)
+    let chatData = game.wfrp4e.utility.chatDataSetup(msg, "gmroll", false);
+    chatData.speaker = { alias: this.name };
+    ChatMessage.create(chatData);
   }
 
+  /**
+   * Finishes disease and handles Lingering symptoms.
+   *
+   * @param {ItemWfrp4e} disease
+   *
+   * @return {Promise<void>}
+   */
   async finishDisease(disease) {
-
     let msg = game.i18n.format("CHAT.DiseaseFinish", { disease: disease.name })
 
-    if (disease.system.symptoms.includes("lingering")) {
+    if (disease.system.symptoms.value.includes("lingering")) {
```

Comment:
Fixing error that happened every time:
![image](https://github.com/moo-man/WFRP4e-FoundryVTT/assets/11304207/bba2f0e5-6f77-4f9d-96b2-b0bb8d230689)